src/foam/util/AddressUtil.java

```diff
@@ -57,4 +66,42 @@
     return new String[] { streetNumber, streetName };
   }
 
+  public static String normalizeRegion(X x, String country, String regionCode) {
+    String normalizedRegion = country + "-" + regionCode;
+    DAO regionDAO = (DAO) x.get("regionDAO");
+    ArraySink sink = (ArraySink) regionDAO.where(AND(
+      EQ(Region.COUNTRY_ID, country)
+    )).select(new ArraySink());
+
+    List<Region> regions = sink.getArray();
+
+    for ( int i = 0; i < regions.size(); i++ ) {
+      Region region = (Region) regions.get(i);
+      if ( region.getIsoCode().equals(regionCode) || region.getName().equalsIgnoreCase(regionCode) ) {
+        normalizedRegion = region.getCode();
+        break;
+      }
+    }
+    return normalizedRegion;
+  }
+
+  public static String normalizeCountry(X x, String countryCode) {
+    String normalizedCountry = countryCode;
+    DAO countryDAO = (DAO) x.get("countryDAO");
+    ArraySink sink = (ArraySink) countryDAO.select(new ArraySink());
```

Comment:
Realize countryDAO is small but why not an OR mlang?

Reply:
also updated to use CONTAINS_IC